SSINS/sky_subtract.py

```diff
@@ -37,29 +37,49 @@ def read(self, filename, diff=False, flag_choice=None, INS=None, custom=None,
         Args:
             filename (str or list of str): The filepath(s) to read in.
             diff (bool): If True, and data was read in, then difference the visibilities in time
+            diff_freq (bool): If True, and data was read in, then difference the visibilities in freq
             flag_choice: Sets flags for the data array on read using apply_flags method.
             INS: An INS object for apply_flags()
             custom: A custom flag array for apply_flags()
             kwargs: Additional kwargs are passed to UVData.read()
+            override_keyword (str): sets a keyword to override to True regardless of internal diffs.
         """
-        warnings.warn("SS.read will be renamed to SS.read_data soon to avoid"
-                      " conflicts with UVData.read.", category=PendingDeprecationWarning)
 
         super().read(filename, **kwargs)
 
+        # always set extra keywords, else key errors when trying to check
         if (self.data_array is not None):
             if diff:
                 self.diff()
                 self.apply_flags(flag_choice=flag_choice, INS=INS, custom=custom)
-            else:
+                self.extra_keywords['dif_time'] = True
```

Comment:
It looks like if I set `diff=False` and `diff_freq=True`, I will end with with both extra keywords `dif_time` and `dif_freq` set to `True`, but I don't think that's what we want. Is this related to the comment above? Shouldn't we default to `False` instead?

Reply:
Added some tests in a231fb6 to affirm that this is not the case.

Reply:
Failed to read the git diff properly. Thanks for adding the unit test.